core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,475 @@
+     * */
+    val rightmostSlash = path.lastIndexOf('/')
+    val rightmostDot = path.lastIndexOf('.')
+    return if (rightmostSlash > -1 && rightmostDot > -1) {
+      path.substring(
+        rightmostSlash + 1, if (rightmostDot > rightmostSlash) rightmostDot else path.length
+      )
+    } else ""
+    // If couldn't find a dot and/or slash in the url
+  }
+
+  // Override onBackPressed() to respond to user pressing 'Back' button on navigation bar
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    return object : Dialog(requireContext(), theme) {
+      override fun onBackPressed() {
+        exitAddNoteDialog()
+      }
+    }
+  }
+
+  private fun exitAddNoteDialog() {
+    if (noteEdited) {
+      alertDialogShower?.show(
+        KiwixDialog.NotesDiscardConfirmation,
+        ::dismissAddNoteDialog
+      )
+    } else {
+      // Closing unedited note dialog straightaway
+      dismissAddNoteDialog()
+    }
+  }
+
+  private fun disableMenuItems() {
+    if (toolbar.menu != null) {
+      val saveItem = toolbar.menu.findItem(R.id.save_note)
+      val shareItem = toolbar.menu.findItem(R.id.share_note)
+      val deleteItem = toolbar.menu.findItem(R.id.delete_note)
+      saveItem.isEnabled = false
+      shareItem.isEnabled = false
+      deleteItem.isEnabled = false
+      saveItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+      shareItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+      deleteItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableSaveNoteMenuItem() {
+    if (toolbar.menu != null) {
+      val saveItem = toolbar.menu.findItem(R.id.save_note)
+      saveItem.isEnabled = true
+      saveItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableDeleteNoteMenuItem() {
+    if (toolbar.menu != null) {
+      val deleteItem = toolbar.menu.findItem(R.id.delete_note)
+      deleteItem.isEnabled = true
+      deleteItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableShareNoteMenuItem() {
+    if (toolbar.menu != null) {
+      val shareItem = toolbar.menu.findItem(R.id.share_note)
+      shareItem.isEnabled = true
+      shareItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  override fun onViewCreated(
+    view: View,
+    savedInstanceState: Bundle?
+  ) {
+    super.onViewCreated(view, savedInstanceState)
+    toolbar.setTitle(R.string.note)
+    toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
+    toolbar.setNavigationOnClickListener {
+      exitAddNoteDialog()
+      closeKeyboard()
+    }
+    toolbar.setOnMenuItemClickListener { item: MenuItem ->
+      when (item.itemId) {
+        R.id.share_note -> { // Opens app-chooser for sharing the note text file
```

Comment:
the brackets aren't necessary on the RHS of the `->`, comments could probably be deleted too